libfreerdp/core/transport.c

```diff
@@ -58,15 +58,158 @@
 
 static void* transport_client_thread(void* arg);
 
+#ifdef WITH_KRB5_SUPPORT
+
+#include <krb5.h>
+#include <winpr/library.h>
+
+static HMODULE g_KRB5Module = NULL;
+
+struct _KRB5FunctionTable
+{
+	krb5_error_code (* pKrb5_init_context)(krb5_context* m_context);
+	krb5_error_code (*pKrb5_parse_name_flags)(krb5_context m_context, const char* m_address, int m_flags, krb5_principal* m_principal);
+	krb5_error_code (*pKrb5_cc_cache_match)(krb5_context m_context, krb5_principal m_client, krb5_ccache * m_id);
+	krb5_error_code (*pKrb5_cc_default)(krb5_context m_context, krb5_ccache * m_id);
+	krb5_error_code (*pKrb5_init_creds_init)(krb5_context m_context, krb5_principal m_client, krb5_prompter_fct m_prompter, void * m_prompter_data, krb5_deltat m_start_time, krb5_get_init_creds_opt * m_options, krb5_init_creds_context * m_ctx);
+	krb5_error_code (*pKrb5_init_creds_set_password)(krb5_context m_context, krb5_init_creds_context m_ctx, const char * m_password);
+	krb5_error_code (*pKrb5_init_creds_get)(krb5_context m_context, krb5_init_creds_context m_ctx);	
+	void (*pKrb5_init_creds_free)(krb5_context m_context, krb5_init_creds_context m_ctx);
+	void (*pKrb5_free_context)(krb5_context m_context);
+};
+typedef struct _KRB5FunctionTable KRB5FunctionTable;
+
+static KRB5FunctionTable g_KRB5 = { 0 };
+
+UINT32 transport_krb5_check_account(char* username, char* domain, char* passwd)
+{
+	krb5_error_code ret;
+	krb5_context context;
+    	krb5_principal principal = NULL;
+	char address[256];
+    	krb5_ccache ccache;
+	krb5_init_creds_context ctx = NULL;
+	
+	memset(address, 0, sizeof(address));
+	
+	strcpy(address, username);
+	strcat(address, "@");
+	strcat(address, domain);
+ 
+	g_KRB5Module = LoadLibrary("libkrb5.so.26");
```

Comment:
If used as library this and the following initialization code might get executed multiple times (maybe also by different threads).
I'd suggest to use InitOnceExecuteOnce (for example like done in winpr/libwinpr/pool/pool.c).

What was initially the idea/reason behind loading the library later on instead of directly linking against it?